cpp/src/db/db_object.cu

```diff
@@ -120,84 +120,62 @@ bool db_pattern<idx_t>::isAllConstants() {
 template class db_pattern<int32_t>;
 template class db_pattern<int64_t>;
 
-template<typename idx_t>
-void db_column_index<idx_t>::deleteData() {
-  if (offsets != nullptr) {
-    ALLOC_FREE_TRY(offsets, nullptr);
-    offsets = nullptr;
-    offsets_size = 0;
-  }
-  if (indirection != nullptr) {
-    ALLOC_FREE_TRY(indirection, nullptr);
-    indirection = nullptr;
-    indirection_size = 0;
-  }
-}
-
 template<typename idx_t>
 db_column_index<idx_t>::db_column_index() {
-  offsets = nullptr;
   offsets_size = 0;
-  indirection = nullptr;
   indirection_size = 0;
 }
```

Comment:
db_column_index() = default; should be sufficient if you initialize offset_size and indirection_size on definition.